File: KTLT_022/main.py
n, k = map(int, input().split())
dp = [0] + list(map(int, input().split()))

sliding_window = [0] * (n + 1)


def solve_sliding_window():
    for i in range(1, n + 1):
        sliding_window[i] = sliding_window[i - 1] + dp[i]

    left = 1
    min_len = 1e9
    for right in range(1, n + 1):
        ans_temp = sliding_window[right] - sliding_window[left - 1]
        while left <= right and ans_temp >= k:
            min_len = min(min_len, right - left + 1)
            left += 1
            ans_temp = sliding_window[right] - sliding_window[left - 1]
    return min_len if min_len != 1e9 else -1


print(solve_sliding_window())

File: KTLT_022/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define ld long double
#define bl boolean
#define str string
#define READ(name)                                                             \
  freopen(name ".inp", "r", stdin);                                            \
  freopen(name ".out", "w", stdout);

int main() {
  READ("1");
  ll n, k;
  cin >> n >> k;
  vector<ll> dp(n + 5, 0);
  for (ll i = 1; i <= n; i++) {
    ll temp;
    cin >> temp;
    (i == 1) ? dp[i] = temp : dp[i] = dp[i - 1] + temp;
  }
  ll left = 1, min_len = 1e9, check_solve = false;
  for(ll right=1;right<=n;right++)
  {
      ll num = dp[right]-dp[left-1];
      while(left<=right&&num>=k)
      {
          min_len=min(min_len,right-left+1),check_solve=true;
          left++;
          num=dp[right]-dp[left-1];
      }
  }
 (check_solve==true)?cout<<min_len:cout<<"-1";
 
}
